### PR TITLE
Feature/favorite functionality

### PR DESCRIPTION
**Pull request by dkwon1223**
# Pull Request 

## Description of Changes

Brings in favorite functionality with drinks allowing users to click the heart icon within the detailed drink view in order to add a drink to their favorites. Also updates mock data to utilize uploaded cocktail image urls and new drinks.

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🔮 New feature (non-breaking change which adds functionality)
- [ ] 📖 Documentation Update
- [ ] Other (please describe):

## Has this been Tested? 🧪

Testing will now begin on all components.

## Checklist ✅

- [x] I have performed a self-review of my own code.
- [x] I have made notes of my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove the new feature works if necessary.
- [ ] All tests are passing.

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -28,6 +28,13 @@ function App() {
     chooseRecipes(filteredRecipes)
   }
 
+  const toggleFavorite = (id) => {
+    const targetDrink = mixedDrinkRecipes.find((drink) => {
+      return drink.id === id;
+    })
+    targetDrink.isFavorite = !targetDrink.isFavorite;
+  }
+
```
Comment (lalonggone) [inline src/components/App/App.js]: awesome work with the toggle. easy to understand and clean!

File: src/components/DrinkDetails/DrinkDetails.js [hunk 2]
```diff
@@ -26,23 +30,22 @@ function DrinkDetails({ toggleFavorite }) {
...
               })}
             </div>
             <div className='drink-details-directions'>
               <h3>Directions</h3>
               <ol>
-                {Object.values(targetDrink.directions).map((value) => {
-                  return <li>{value}</li>
+                {Object.values(targetDrink.directions).map((value, index) => {
+                  return <li key={index}>{value}</li>
                 })}
```
Comment (lalonggone) [inline src/components/DrinkDetails/DrinkDetails.js:42]: don't forget the key!! yeah!!

File: src/components/NavBar/NavBar.js [hunk 3]
```diff
@@ -22,13 +22,15 @@ function NavBar({ setDrinkType }) {
...
   return (
     <div>
       <div className="border"></div>
       <nav className="nav-container">
         <ul className="nav-list">
-          <li className="nav-item" onClick={refreshPage} >
-            <Link to="/random" >
+          <li className="nav-item">
+            <Link to={`/drink/${randomId}`} >
               Random Drink
```
Comment (lalonggone) [inline src/components/NavBar/NavBar.js:34]: I love this approach to get a random drink! thanks for implementing that

**Review by lalonggone — COMMENTED** [review]
I noticed the grid styling doesn't center "leftover" cards on the next row or when there aren't enough to fill the width of the page. Maybe something to look into if we have time. but otherwise this feature looks GREAT! nice work on the Favorite functionality and all of that styling

File: src/components/DrinkDetails/DrinkDetails.js [hunk 4]
```diff
@@ -1,22 +1,27 @@
...
     return drink.id === parsedId;
   })
   return (
     <section className='drink-details-page'>
-      <NavBar className='drink-details-nav'/>
+      <NavBar className='drink-details-nav' setDrinkType={setDrinkType}/>
       <div className='drink-details-container'>
         <img src={targetDrink.image} alt={`aesthetic view of a ${targetDrink.name}`} className='drink-details-image'/>
-        <img src={HeartIcon} alt="heart icon" className='drink-details-heart-icon'/>
+        <img src={targetDrink.isFavorite ? HeartIconRed : HeartIcon} alt="heart icon" className='drink-details-heart-icon' onClick={() => {
```
Comment (crotteau) [inline src/components/DrinkDetails/DrinkDetails.js:21]: nice use of the ternary operator

**Review by crotteau — APPROVED** [review]

**Comment by crotteau** [comment]
Everything looks good, we could try adding some more semantic html into the drink detail component

**Pull request merged**